backend/molgenis-emx2-beacon-v2/src/main/java/org/molgenis/emx2/beaconv2/vcfimport/VcfImport.java

```diff
@@ -0,0 +1,78 @@
+package org.molgenis.emx2.beaconv2.vcfimport;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFileReader;
+import htsjdk.variant.vcf.VCFHeader;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.molgenis.emx2.Row;
+import org.molgenis.emx2.Table;
+
+public class VcfImport {
+
+  private File vcfTmpFile;
+  private Table genomicVariations;
+  private Table individuals;
+
+  public VcfImport(File vcfTmpFile, Table genomicVariations, Table individuals) {
+    this.vcfTmpFile = vcfTmpFile;
+    this.genomicVariations = genomicVariations;
+    this.individuals = individuals;
+  }
+
+  public int start() {
+
+    VCFFileReader vcfFileReader = new VCFFileReader(vcfTmpFile, false);
+    // Retrieve the header
+    VCFHeader header = vcfFileReader.getFileHeader();
+    System.out.println("VCF Header: " + header);
+
+    int count = 0;
+    List<Row> rows = new ArrayList<>();
+    for (VariantContext variant : vcfFileReader) {
+
+      Row row = new Row();
+      row.set("variantInternalId", UUID.randomUUID());
```

Comment:
so this would then need to be an adapter. I.e. using the 'strategy' design pattern where the way on how to handle rows from the VCF file can have different implementations.

Reply:
Sounds good, but this needs to be fully transparant to the user so I think we need to make https://github.com/molgenis/GCC/issues/775 into a more general GUI tool